test/remote-github-patch-text.ts

```diff
@@ -0,0 +1,454 @@
+    sandbox.assert.calledOnceWithExactly(stub, {
+      owner: upstream.owner,
+      repo: upstream.repo,
+      pull_number: pullNumber,
+      per_page: pageSize,
+    });
+  });
+  it('Returns all the valid patches', async () => {
+    // setup
+    const listFilesOfPRResult = {
+      headers: {},
+      status: 200,
+      url: 'http://fake-url.com',
+      data: [
+        {
+          sha: 'a1d470fa4d7b04450715e3e02d240a34517cd988',
+          filename: 'Readme.md',
+          status: 'modified',
+          additions: 4,
+          deletions: 1,
+          changes: 5,
+          blob_url:
+            'https://github.com/TomKristie/HelloWorld/blob/eb53f3871f56e8dd6321e44621fe6ac2da1bc120/Readme.md',
+          raw_url:
+            'https://github.com/TomKristie/HelloWorld/raw/eb53f3871f56e8dd6321e44621fe6ac2da1bc120/Readme.md',
+          contents_url:
+            'https://api.github.com/repos/TomKristie/HelloWorld/contents/Readme.md?ref=eb53f3871f56e8dd6321e44621fe6ac2da1bc120',
+          patch:
+            '@@ -1,2 +1,5 @@\n Hello world\n-!\n+Goodbye World\n+gOodBYE world\n+\n+Goodbye World',
+        },
+        {
+          sha: '8b137891791fe96927ad78e64b0aad7bded08bdc',
+          filename: 'foo/foo.txt',
+          status: 'modified',
+          additions: 1,
+          deletions: 1,
+          changes: 2,
+          blob_url:
+            'https://github.com/TomKristie/HelloWorld/blob/eb53f3871f56e8dd6321e44621fe6ac2da1bc120/foo/foo.txt',
+          raw_url:
+            'https://github.com/TomKristie/HelloWorld/raw/eb53f3871f56e8dd6321e44621fe6ac2da1bc120/foo/foo.txt',
+          contents_url:
+            'https://api.github.com/repos/TomKristie/HelloWorld/contents/foo/foo.txt?ref=eb53f3871f56e8dd6321e44621fe6ac2da1bc120',
+          patch: '@@ -1 +1 @@\n-Hello foo\n+',
+        },
+        {
+          sha: '3b18e512dba79e4c8300dd08aeb37f8e728b8dad',
+          filename: 'helloworld.txt',
+          status: 'removed',
+          additions: 0,
+          deletions: 1,
+          changes: 1,
+          blob_url:
+            'https://github.com/TomKristie/HelloWorld/blob/f5da827a725a701302da7db2da16b1678f52fdcc/helloworld.txt',
+          raw_url:
+            'https://github.com/TomKristie/HelloWorld/raw/f5da827a725a701302da7db2da16b1678f52fdcc/helloworld.txt',
+          contents_url:
+            'https://api.github.com/repos/TomKristie/HelloWorld/contents/helloworld.txt?ref=f5da827a725a701302da7db2da16b1678f52fdcc',
+          patch: '@@ -1 +0,0 @@\n-hello world',
+        },
+      ],
+    };
+    sandbox.stub(octokit.pulls, 'listFiles').resolves(listFilesOfPRResult);
+
+    // tests
+    const {patches, filesMissingPatch} = await getCurrentPulLRequestPatches(
+      octokit,
+      upstream,
+      pullNumber,
+      pageSize
+    );
+    expect(patches.size).equals(3);
+    expect(patches.get(listFilesOfPRResult.data[0].filename)).equals(
+      listFilesOfPRResult.data[0].patch
```

Comment:
We should explicitly put the expected data here - not a reference into the input data. This can lead to invalid tests where the test passes, but the data in incorrect.